flink-runtime/src/main/java/org/apache/flink/runtime/metrics/ReporterSetup.java

```diff
@@ -120,55 +124,37 @@ public static ReporterSetup forReporter(String reporterName, MetricConfig metric
 	}
 
 	private static ReporterSetup createReporterSetup(String reporterName, MetricConfig metricConfig, MetricReporter reporter) {
-		LOG.info("Configuring {} with {}.", reporterName, metricConfig);
+		LOG.debug("Configuring {} with {}.", reporterName, metricConfig);
```

Comment:
we should actually remove this line (or exclude the metricConfig) as we may be leaking sensitive information.

Reply:
Good point. I am not 100% sure about it - do we have a Flink-wide way to handle such cases? I guess the cleanest approach would be to have a special set of keys that are considered sensitive, which have to be obfuscated prior to logging. Seeing which other config values are used during the initialization could be generally pretty useful for debugging.

Reply:
Ping @AHeise @zentol - could we agree on something here?
I am always "pro extensive logging", but this could be professional deformation. Being able to "on-demand" see what is going on is very valuable for production systems. We could declare somewhere for Flink in general that if you choose to run with debug log level, some potentially sensitive information could leak into logs. My arguments are:
1) If someone has uncontrolled access to the log files on your machine in production, content of this file is probably not the biggest of your problems.
2) Running with debug level is not a "normal" scenario - this is intended for hands on investigation of issues. Log level for potentially compromisable external systems could be explicitly set to trace in such cases.
3) We have been "leaking" this data in the current versions with info (!) level without much concern

Reply:
> do we have a Flink-wide way to handle such cases

The `GlobalConfiguration` contains a set of keys that are considered sensitive, which we use for the WebUI and various INFO logging.

> We have been "leaking" this data in the current versions with info (!) level without much concern

Few reporters actually use credentials (afaik only datadog does), so the sample size is fairly low.

> If someone has uncontrolled access to the log files on your machine in production, content of this file is probably not the biggest of your problems.

Doesn't need access to the machine; access to the UI is sufficient, which was grave enough that we introduced the whole secret-key concept in the first place.

> Running with debug level is not a "normal" scenario - this is intended for hands on investigation of issues. Log level for potentially compromisable external systems could be explicitly set to trace in such cases.

There's precedence with FLINK-10363 that credentials should not be logged even on debug.
FLINK-16478 also which proposes a REST API for modifying the log level potentially voiding any argument for it being opt-in insecurity.

I would approach this cautiously and never log anything sensitive.

Reply:
Removed.